### DISCUSSION TITLE
ShokaX 下一代的技术选型

### DISCUSSION THREAD

**Discussion by zkz098** [theme-shoka-x/hexo-theme-shokaX#165] [Polls]
本讨论主要决定未来很长一段时间内 ShokaX 的基础技术框架，目前提供两个可选项:
- Vue: 优势在于可以和现有的部分博客系统快速集成（例: valaxy 和 vitepress），且学习曲线较为平滑，对部分非前端领域的开发者来说比较容易上手。
- React: 优势在于和 ShokaX 相性较好，相对容易迁移。

请注意：本投票不一定为最终结论，受制于开发人员经验等问题，最终结果可能存在变化。
如果您未来愿意加入 ShokaX 开发组，我建议您选择一个您更擅长（或您认为更好）的框架

如果有更好的选择，可在评论区提出

**Comment by D-Sketon** [comment]
要不两个都做吧（）

**Comment by MisakaMikoto521** [comment]
要不两个都做吧（）

**Comment by zkz098** [comment]
啊？
基础设施整一个 Vue 和 React 双版本的好像意义不大，再等等，应该能分出来（

**Comment by zkz098** [comment]
寄，平票（
整Vue的可能更好，实在不行还能后撤到Vitepress/Valaxy，不至于迁完astro没人用

**Reply by zkz098** [reply]
但对于SX这种陈年老项目可能React更河里（）
要不然再摇点人投票（）

**Comment by D-Sketon** [comment]
现在是已经开始迁移了吗（）

**Reply by zkz098** [reply]
如迁（）
技术选型确定不下来，现在迁移容易出逝故

**Reply by D-Sketon** [reply]
> 如迁（）
> 技术选型确定不下来，现在迁移容易出逝故

感觉还好
Vue和React之间的互相迁移感觉也没那么复杂（）

**Reply by zkz098** [reply]
那就先按照 Vue 迁，如果 Astro 出现没人用的问题还能后撤到 Valaxy/Vitepress，不至于完全寄掉（）

**Comment by zkz098** [comment]
迁移工作将于近期开始，迁移目标系统为 Valaxy
争取2025年底前发布第一个测试版（）
